swatch-utilization/src/main/java/com/redhat/swatch/utilization/service/CustomerOverUsageService.java

```diff
@@ -107,8 +119,30 @@ public void check(UtilizationSummary payload) {
    */
   private void checkMeasurement(
       UtilizationSummary payload, Measurement measurement, Double threshold) {
+    // Skip unlimited capacity measurements
+    if (Boolean.TRUE.equals(measurement.getUnlimited())) {
+      log.debug(
+          "Skipping over-usage check for unlimited capacity: orgId={} productId={} metricId={}",
+          payload.getOrgId(),
+          payload.getProductId(),
+          measurement.getMetricId());
+      return;
+    }
+
     double currentTotal = measurement.getCurrentTotal();
     double capacity = measurement.getCapacity();
+
+    // Skip invalid or zero capacity measurements
+    if (capacity <= 0.0) {
```

Comment:
Have to check pure payg contract payload without any capacity. Since I am expecting this will be 0 for them and could be a valid scenario.